### DISCUSSION TITLE
results, list index of out of range?

### DISCUSSION THREAD

**Discussion by apiszcz** [ultralytics/ultralytics#8436] [Q&A]
When using the tracker I get an error:

`results = self.md[model].track(source=rowd['frames'], verbose=False, device=0, stream=False, persist=True, tracker=pathtotrackerconifg)

*** IndexError: list index out of range
`

Using predict works correctly

`results = self.md[model].predict(source=rowd['frames'], verbose=False, device=y.gpu)
`

**Comment by glenn-jocher** [comment]
Hi there! It seems like the issue might be related to the `source` input or the tracker configuration. Could you please double-check that `rowd['frames']` contains a valid list of frames or a path to a video file? Also, ensure that `pathtotrackerconifg` points to a correct tracker configuration file. If the problem persists, could you share a minimal code snippet that reproduces the error, along with the versions of the relevant libraries you're using? This will help us diagnose the issue more effectively. 😊🔍

For example, your tracking code should look something like this:

```python
results = self.md[model].track(source='path/to/video.mp4', device=0, stream=False, persist=True, tracker='path/to/tracker.yaml')
```

And ensure that the tracker configuration file exists at the specified path:

```python
import os
assert os.path.isfile('path/to/tracker.yaml'), 'Tracker config file not found'
```

Let's get this sorted out! 🛠️

**Comment by apiszcz** [comment]
Hi, 

1 I am using a single image, also tried a list with a single image as the source, not a video file.
<PIL.Image.Image image mode=RGB ....
2 ls -l to the tracker configuration file is ok. Verified.
3 test with predict, and (img, verbose=False)
same error as below.

Updated the systems to supervision 0.18 and to the latest yolo 8.1.18. Example is not using or trying to access obb attributes. Using .track or predict. I'm curious as to the code fragment where core.py is trying to access .obb in a list type object. It seems like a dictionary is expected. Tested with .predict, same error. Is it possible I am using supervision incorrectly? Same code works with earlier versions of yolo (8.0) and supervision.

detections = supervision.Detections.from_ultralytics(result)

> test/lib/python3.10/site-packages/supervision/detection/core.py(147)from_ultralytics()
-> @classmethod
(Pdb) s
> test/lib/python3.10/site-packages/supervision/detection/core.py(181)from_ultralytics()
-> if ultralytics_results.obb is not None:
(Pdb) ultralytics_results.obb
*** AttributeError: 'list' object has no attribute 'obb'


(Pdb) type(ultralytics_results)
<class 'list'>
FILE: supervision/detection/core.py

        if ultralytics_results.obb is not None:
            return cls(


=== display of 'result'
```
[[ultralytics.engine.results.Results object with attributes:                                                                                                                                                                                                                                                                                                                            boxes: ultralytics.engine.results.Boxes object                                                                                                                                              keypoints: None                                                                                                                                                                             
masks: None                                                                                                                                                                                 
names: {0:
....
obb: None                                                                                                                                                                                   
orig_img: array([[[  
....
orig_shape: (256, 256)
path: 'image0.jpg'
probs: None
save_dir: None
speed: {'preprocess': 18.550872802734375, 'inference': 195.73092460632324, 'postprocess': 166.5358543395996}]]
```

```

(Pdb) type(result[0])
<class 'ultralytics.engine.results.Results'>

(Pdb) result[0].obb is None
True
```

```
(Pdb) supervision.Detections.from_ultralytics(result)
*** AttributeError: 'list' object has no attribute 'obb'

```

this works as expected, my error ... 
`supervision.Detections.from_ultralytics(result[0])`


Thank you.

**Comment by apiszcz** [comment]
resolved
https://github.com/orgs/ultralytics/discussions/8436

**Reply by glenn-jocher** [reply]
I'm glad to hear the issue is resolved! If you have any more questions or run into other issues in the future, don't hesitate to reach out. Happy coding! 😊🚀
